jpa/org.eclipse.persistence.jpa/src/org/eclipse/persistence/internal/jpa/metadata/accessors/objects/MetadataAnnotatedElement.java

```diff
@@ -202,17 +204,29 @@ public MetadataAnnotation getAnnotation(Class annotation) {
     /**
      * INTERNAL:
      * Return the annotated element for this accessor. Note: This method does
-     * not check against a metadata complete.
+     * not check against a meta-data complete.
+     * @param annotation annotation name
      */
-    public MetadataAnnotation getAnnotation(String annotation) {
+    public MetadataAnnotation getAnnotation(final String annotation) {
+        return getAnnotation(annotation, (Set<String>)null);
```

Comment:
why not pass new HashSet here and replace null check with isEmpty() later?

Reply:
I don't see significant difference in current code or your proposal. Some check still needs to be there (and I think null check may be a bit faster than calling a method of Hashset). I don't care much about this. If you think that your option will look better, I'll put it there.